### PR TITLE
Implement wrapper for images 

### PR DESCRIPTION
**Pull request by fredrikved**
Implement a wrapper for images that need a token.

### CODE REVIEW

File: src/components/ProtectedImage.vue [hunk 1]
```diff
@@ -0,0 +1,26 @@
+<script lang="ts" setup>
+import filters from "@/utils/filters";
+import { useAuth0 } from "@auth0/auth0-vue";
+import { onMounted, ref } from "vue";
+
+const props = defineProps<{
+  src: string | null | undefined;
```
Comment (SimonSimCity) [inline src/components/ProtectedImage.vue:7]: Would it be better not to allow `null` or `undefined` here? There won't be an image to show in that case anyways and the template including this one would have to take care about this (either not showing the image or using a placeholder or whatever fits the case).

File: src/components/ProtectedImage.vue [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+}>();
+
+const source = ref("");
+
+// TODO: fix this for tests (auth0 isn't in context in tests at this time)
```
Comment (SimonSimCity) [inline src/components/ProtectedImage.vue:12]: Should be fairly easy to do using a `spy` as I've done in `PlaylistOverview.spec.ts`

**Review by SimonSimCity — CHANGES_REQUESTED** [review]

**Comment by SimonSimCity** [comment]
@kkuepper any idea why we don't get a code coverage report? 🤔

**Review by kkuepper — APPROVED** [review]

File: src/components/ProtectedImage.vue (hunk 1 above)
Comment (fredrikved) [reply src/components/ProtectedImage.vue:7]: I guess implementing the placeholder outside is better. Originally thought of doing that internally in the component.

**Comment by SimonSimCity** [comment]
Thanks for adding the test and fixing the outstanding comments 😃 I'll merge 👍

**Review by SimonSimCity — APPROVED** [review]

**Pull request merged**